### PR TITLE
omprog/omstdout: guard empty template LF checks

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary

This hardens `omprog` and `omstdout` for the case where a configured template renders to an empty string.

Both modules used the rendered length to inspect the last byte before appending a trailing newline. For an empty rendered template, that check indexed one byte before the template buffer. Empty template output is valid input to an action, so the modules should treat it like other non-newline-terminated output without reading before the buffer.

## Changes

- Guard `omprog` and `omstdout` trailing-newline checks with `len == 0`.
- Preserve the existing behavior of appending a newline when needed.
- Add regression tests for empty rendered templates reaching each module.
- Use `set_proper_termination_file` / `check_proper_termination` so sanitizer builds catch the original abort and normal CI seals clean shutdown.

## Validation

Initial reproducer before the fix:

- `omstdout`: ASan reported a one-byte heap-buffer-under-read in `plugins/omstdout/omstdout.c` at the trailing-LF check.
- `omprog`: ASan reported the same pattern in `plugins/omprog/omprog.c` at the trailing-LF check.

Local validation after the fix:

- `devtools/format-code.sh --git-changed --check --check-if-available`
- `devtools/check-test-antipatterns.sh tests/omstdout-empty-template.sh tests/omprog-empty-template.sh`
- `shellcheck -S warning tests/omstdout-empty-template.sh tests/omprog-empty-template.sh`
- `./devtools/local-validation-plan.sh --base upstream/main --run`
  - Ubuntu 26.04 dev container: `rsyslog/rsyslog_dev_base_ubuntu:26.04`, digest `sha256:32ade478a405e4f27f077b5268ec5ecc59dd572843ad67ca2b6723594960ae09`
  - full container test summary observed: `TOTAL: 1247`, `PASS: 1214`, `SKIP: 33`, `FAIL: 0`
  - local Cubic review: no issues found
- Post-rebase focused ASan validation:
  - `make -C tests check TESTS="omstdout-empty-template.sh omprog-empty-template.sh"`
  - `TOTAL: 2`, `PASS: 2`, `FAIL: 0`

<a href="https://cubic.dev/pr/rsyslog/rsyslog/pull/7310?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: plugins/omstdout/omstdout.c [hunk 1]
```diff
@@ -206,7 +206,10 @@ BEGINdoAction
     if ((r = write(1, toWrite, len)) != (int)len) { /* 1 is stdout! */
         DBGPRINTF("omstdout: error %d writing to stdout[%zd]: %s\n", r, len, toWrite);
     }
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply plugins/omstdout/omstdout.c]: Done, guarded the initial omstdout write so the empty-template path skips the redundant zero-length write and still appends the configured newline.
